### PR TITLE
PHP 8.0 compatibility

### PR DESCRIPTION
**Pull request by jfernancordova**
## Description and Background

PHP 8.0 compatibility

## Motivation and Context

Issue #162  
It's time to slowly check compatibility for PHP 8.0

## How Has This Been Tested

- Change composer.json removing  PHP version less than 7.3
- Upgrading phpunit to ^9.0
- Change travis.yml removing  PHP version less than 7.3
- Change rector.yml removing  PHP version less than 7.3

#### Checklist

- [x] My code follows the PSR coding guideline.
- [x] I have updated the `documentation` accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added `tests` to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ you can execute any phpsu command via something like this:
 
 The following versions of PHP are supported by this version.
 
-* PHP `7.2`, `7.3`, `7.4`
+* PHP `7.3`, `7.4`
```
Comment (ChrisB9) [inline README.md:60]:
```suggestion
* PHP `7.3`, `7.4`, `8.0`
```

**Review by ChrisB9 — CHANGES_REQUESTED** [review]

**Review by ChrisB9 — APPROVED** [review]

**Comment by ChrisB9** [comment]
@jfernancordova Thank you for your effort, we will merge this on friday.

File: README.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ you can execute any phpsu command via something like this:
 
 The following versions of PHP are supported by this version.
 
-* PHP `7.2`, `7.3`, `7.4`
+* PHP `7.3`, `7.4`, `8.0`
```
Comment (ChrisB9) [inline README.md:60]:
```suggestion
* PHP `7.2`, `7.3`, `7.4`, `8.0`
```

File: composer.json [hunk 3]
```diff
@@ -49,7 +49,7 @@
   "require": {
     "symfony/console": "^4.0 | ^5.0",
     "symfony/process": "^4.0 | ^5.0",
-    "php": ">=7.2",
+    "php": "^7.3 || ~8.0.0",
```
Comment (ChrisB9) [inline composer.json:52]:
```suggestion
    "php": "^7.2 || ~8.0.0",
```

File: composer.json [hunk 4]
```diff
@@ -60,7 +60,7 @@
     "phpstan/phpstan": "^0.12.18",
     "infection/infection": "^0.15.3",
     "phpunit/php-invoker": "^2.0 | ^3.0",
-    "phpunit/phpunit": "^8.5"
+    "phpunit/phpunit": "^9.0"
```
Comment (ChrisB9) [inline composer.json:63]:
```suggestion
    "phpunit/phpunit": "^8.5 || ^9.0"
```

File: rector.yml [hunk 5]
```diff
@@ -2,9 +2,8 @@ parameters:
   sets:
     - 'code-quality'
     - 'dead-code'
-    - 'php72'
     - 'php73'
     - 'php74'
-  php_version_features: '7.2'
+  php_version_features: '7.3'
```
Comment (ChrisB9) [inline rector.yml:7]:
```suggestion
  php_version_features: '7.2'
```

**Review by ChrisB9 — CHANGES_REQUESTED** [review]
We realized we need PHP 7.2 support and can't drop it yet. We are going to drop support for 7.2 (and maybe 7.3) with the next major version

**Comment by jfernancordova** [comment]
> We realized we need PHP 7.2 support and can't drop it yet. We are going to drop support for 7.2 (and maybe 7.3) with the next major version

Ok, I will update adding it again.

File: .travis.yml [hunk 6]
```diff
@@ -1,7 +1,6 @@
 language: php
 
 php:
-  - "7.2"
```
Comment (ChrisB9) [inline .travis.yml:4]: can you also reactivate these? I wasnt able to make a suggestion for them

File: rector.yml [hunk 7]
```diff
@@ -2,7 +2,6 @@ parameters:
   sets:
     - 'code-quality'
     - 'dead-code'
-    - 'php72'
```
Comment (ChrisB9) [inline rector.yml:5]: can you also reactivate these? I wasnt able to make a suggestion for them

**Review by ChrisB9 — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 6 above)
Comment (jfernancordova) [reply .travis.yml:4]: Fixed!

File: rector.yml (hunk 7 above)
Comment (jfernancordova) [reply rector.yml]: Fixed!

**Review by ChrisB9 — APPROVED** [review]

**Pull request merged**